string solve_str
string hand_axis
---
string result
